### ISSUE TITLE
[Test failure] [Daily] [dev12] Legacy build fails unit tests

### ISSUE THREAD

**Issue by dilijev** [Microsoft/ChakraCore#528]
Since the same issues have been there for a while I wanted to open an issue to track.

@nmostafa is investigating and can potentially provide more info on the investigation in this issue.

When these breaks are resolved we can move forward with #330. 

See: https://github.com/Microsoft/ChakraCore/wiki/Build-Status#legacy-builds

**Comment by dilijev** [comment]
@abchatra I don't know whether this should be assigned the 1.2 milestone but @curtisman approves of getting these fixes into master whenever they are available.

**Comment by nmostafa** [comment]
@leirocks is looking into testSimdLink-2.js failures. Recently merged #444 should fix the rest.

**Comment by dilijev** [comment]
@nmostafa Is anyone looking into the PreFAST failures?

**Comment by nmostafa** [comment]
Are you referring to this:
`e:\chakra5\core\lib\runtime\language\dynamicprofileinfo.cpp(214): warning C6011: Dereferencing NULL pointer 'functionBody'.  [e:\nagy\git\chakra5\core\lib\Runtime\Language\Chakra.Runtime.Language.vcxproj]`

We are not looking into this. Not related to SIMD.js

**Comment by dilijev** [comment]
Okay.

Full list of PreFAST warnings for reference:

http://dotnet-ci.cloudapp.net/job/Microsoft_ChakraCore/job/daily_dev12_x64_release/51/consoleFull#1026559514ac1851af-571d-472f-b664-6657b078a5dd

```
20:38:56 d:\j\workspace\daily_dev12_x---5f770fb3\bin\ch\ch.cpp(254): warning C6102: Using 'bcBuffer' from failed function call at line '179'. [d:\j\workspace\daily_dev12_x---5f770fb3\bin\ch\ch.vcxproj]
20:38:56 d:\j\workspace\daily_dev12_x---5f770fb3\bin\ch\ch.cpp(338): warning C6102: Using 'bcBuffer' from failed function call at line '319'. [d:\j\workspace\daily_dev12_x---5f770fb3\bin\ch\ch.vcxproj]
20:38:56 d:\j\workspace\daily_dev12_x---5f770fb3\bin\ch\chakrartinterface.cpp(28): warning C6102: Using 'modulename' from failed function call at line '27'. [d:\j\workspace\daily_dev12_x---5f770fb3\bin\ch\ch.vcxproj]
20:38:56 d:\j\workspace\daily_dev12_x---5f770fb3\bin\ch\helpers.cpp(52): warning C6387: 'file' could be '0':  this does not adhere to the specification for the function 'ftell'. See line 51 for an earlier location where this can occur [d:\j\workspace\daily_dev12_x---5f770fb3\bin\ch\ch.vcxproj]
20:38:56 d:\j\workspace\daily_dev12_x---5f770fb3\bin\ch\helpers.cpp(51): warning C6387: 'file' could be '0':  this does not adhere to the specification for the function 'fseek'.  [d:\j\workspace\daily_dev12_x---5f770fb3\bin\ch\ch.vcxproj]
20:38:56 d:\j\workspace\daily_dev12_x---5f770fb3\bin\ch\helpers.cpp(64): warning C6387: 'file' could be '0':  this does not adhere to the specification for the function 'fread'. See line 51 for an earlier location where this can occur [d:\j\workspace\daily_dev12_x---5f770fb3\bin\ch\ch.vcxproj]
20:38:56 d:\j\workspace\daily_dev12_x---5f770fb3\bin\ch\helpers.cpp(65): warning C6387: 'file' could be '0':  this does not adhere to the specification for the function 'fclose'. See line 51 for an earlier location where this can occur [d:\j\workspace\daily_dev12_x---5f770fb3\bin\ch\ch.vcxproj]
20:38:56 [daily_dev12_x---5f770fb3] $ powershell.exe -NonInteractive -ExecutionPolicy ByPass "& 'C:\Users\DOTNET~1\AppData\Local\Temp\2\hudson7084396863305529629.ps1'"
20:38:56 d:\j\workspace\daily_dev12_x---5f770fb3\bin\ch\helpers.cpp(52): warning C6387: 'file' could be '0':  this does not adhere to the specification for the function 'ftell'. See line 51 for an earlier location where this can occur
20:38:56 d:\j\workspace\daily_dev12_x---5f770fb3\bin\ch\helpers.cpp(51): warning C6387: 'file' could be '0':  this does not adhere to the specification for the function 'fseek'. 
20:38:56 d:\j\workspace\daily_dev12_x---5f770fb3\bin\ch\helpers.cpp(64): warning C6387: 'file' could be '0':  this does not adhere to the specification for the function 'fread'. See line 51 for an earlier location where this can occur
20:38:57 d:\j\workspace\daily_dev12_x---5f770fb3\bin\ch\helpers.cpp(65): warning C6387: 'file' could be '0':  this does not adhere to the specification for the function 'fclose'. See line 51 for an earlier location where this can occur
20:38:57 d:\j\workspace\daily_dev12_x---5f770fb3\lib\common\memory\arenaallocator.cpp(1196): warning C28182: Dereferencing NULL pointer. 'cache' contains the same NULL value as 'curWeakRef' did. See line 1196 for an earlier location where this can occur
20:38:57 d:\j\workspace\daily_dev12_x---5f770fb3\lib\common\memory\arenaallocator.cpp(1276): warning C28182: Dereferencing NULL pointer. 'cache' contains the same NULL value as 'curWeakRef' did. See line 1276 for an earlier location where this can occur
20:38:57 d:\j\workspace\daily_dev12_x---5f770fb3\lib\common\memory\recyclerweakreference.h(233): warning C28182: Dereferencing NULL pointer. 'buckets' contains the same NULL value as 'current' did. 
20:38:57 d:\j\workspace\daily_dev12_x---5f770fb3\lib\jsrt\jsrt.cpp(863): warning C28196: The requirement that '*_Param_(2)!=0' is not satisfied. (The expression does not evaluate to true.)
20:38:57 d:\j\workspace\daily_dev12_x---5f770fb3\lib\runtime\bytecode\bytecodegenerator.cpp(3733): warning C6011: Dereferencing NULL pointer 'invertedStmt'. 
20:38:57 d:\j\workspace\daily_dev12_x---5f770fb3\lib\runtime\language\dynamicprofileinfo.cpp(586): warning C6011: Dereferencing NULL pointer 'callSiteInfo[callSiteId].u.polymorphicCallSiteInfo'. See line 578 for an earlier location where this can occur
```

**Comment by dilijev** [comment]
I'm looking into the PreFAST errors.

**Comment by dilijev** [comment]
I split off the PreFast errors issue to #553

**Comment by dilijev** [comment]
@nmostafa Is your part of this issue fully resolved? Can we close this issue now?

**Comment by nmostafa** [comment]
Yes, #444 resolves the alignment issue, and #549 disables failing Asmjs linking tests (tracked separately in #560, for which I am working on a fix).

**Comment by dilijev** [comment]
Okay, closing this issue as all sub-tasks have been resolved or are being tracked separately.
